### ISSUE TITLE
CastError: Cast to string failed for value

### ISSUE THREAD

**Issue by jaffarc** [nodkz/mongoose-plugin-autoinc#37]
out of nowhere started to give the following error.

I tried to find the error and arrived at     `yield IC.findOneAndUpdate({`. that is receiving an object  in`doc.get(settings.groupingField)`

 mongoose: 5.11.4
 nodejs: 12.20.0
```
CastError: Cast to string failed for value "{
  turma: [],
  status: true,
  infantil: false,
  deletar: false,
  tipo: 58862a828f3e2721f8d920ca
}" at path "groupingField" for model "IdentityCounter"
    at model.Query.exec (/var/www/api-nodejs/node_modules/mongoose/lib/query.js:4358:21)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:17:8
    at Generator.next (<anonymous>)
    at step (/var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:191)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:437
    at new Promise (<anonymous>)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:99
    at createCounterIfNotExist (/var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:55:17)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:64:15
    at Generator.next (<anonymous>)
    at step (/var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:191)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:437
    at new Promise (<anonymous>)
    at /var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:177:99
    at preSave (/var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:125:18)
    at model.<anonymous> (/var/www/api-nodejs/node_modules/mongoose-plugin-autoinc/lib/index.js:313:7)
    at callMiddlewareFunction (/var/www/api-nodejs/node_modules/kareem/index.js:482:23)
    at next (/var/www/api-nodejs/node_modules/kareem/index.js:58:7)
    at Kareem.execPre (/var/www/api-nodejs/node_modules/kareem/index.js:86:8)
    at Kareem.wrap (/var/www/api-nodejs/node_modules/kareem/index.js:265:8)
    at model.$__validate (/var/www/api-nodejs/node_modules/kareem/index.js:375:11)
    at /var/www/api-nodejs/node_modules/mongoose/lib/document.js:2226:10 {
  messageFormat: undefined,
```
can anybody help me

Thanks.

**Comment by Oliver-Weng-Sorcerer** [comment]
I have the same issue, wondering what's the alternative for this one.

**Comment by Pinjasaur** [comment]
+1, also am also seeing this error. cc @nodkz

**Comment by jaffarc** [comment]
I solved it with the version of: "mongoose": "5.10.11" 
```
rm -rf node_modules/
npm i 
```
do not update the mongoose something in the pre save has changed and is passing object
@OliverwengFiltered  @Pinjasaur

**Comment by Pinjasaur** [comment]
> I solved it with the version of: "mongoose": "5.10.11"
> 
> ```
> rm -rf node_modules/
> npm i 
> ```
> 
> do not update the mongoose something in the pre save has changed and is passing object
> @OliverwengFiltered @Pinjasaur

Excellent, seems to have done the trick for me. TYVM @jaffarc! 🎉

**Comment by vkarpov15** [comment]
We fixed this in Mongoose, we had an unintended breaking change in how we handled calling `doc.get()` with an empty string `doc.get('')` in v5.11.0. The fix will be in v5.11.7, which will be released in a few hours. Sorry for the inconvenience!
